akka-stream/src/main/scala/akka/stream/StreamTimeoutException.scala

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (C) 2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream
+
+import scala.concurrent.TimeoutException
+import scala.util.control.NoStackTrace
+
+sealed class StreamTimeoutException(msg: String) extends TimeoutException(msg) with NoStackTrace
```

Comment:
Becomes public from java, so also add:

```suggestion
/**
 * Base class for exceptions specific to Akka Streams
 * 
 * Not for user extension
 */
@DoNotInherit
sealed class StreamTimeoutException(msg: String) extends TimeoutException(msg) with NoStackTrace
```